src/parcsr_ls/par_ilu.c

```diff
@@ -2960,7 +2462,9 @@ hypre_ParILUCusparseSchurGMRESDummySolve( void               *ilu_vdata,
    hypre_Vector            *f_local          = hypre_ParVectorLocalVector(f);
    HYPRE_Real              *f_data           = hypre_VectorData(f_local);
 
-   cudaDeviceSynchronize();
+   /* RL: why need sync */
+   hypre_SyncCudaDevice(hypre_handle());
```

Comment:
Good point, I would say we don't need it, because synchronization is done in the following call to memcpy